aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CreateHandler.java

```diff
@@ -16,35 +18,61 @@
-                        ).toLowerCase());
-                    return ProgressEvent.progress(model, progress.getCallbackContext());
-                })
-                .then(progress -> proxy.initiate("rds::create-db-parameter-group", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
-                        .translateToServiceRequest(resourceModel -> Translator.createDbParameterGroupRequest(resourceModel, Tagging.mergeTags(request.getSystemTags(), request.getDesiredResourceTags())))
-                        .backoffDelay(CONSTANT)
-                        .makeServiceCall((createDBParameterGroupRequest, proxyInvocation) ->
-                                proxyInvocation.injectCredentialsAndInvokeV2(createDBParameterGroupRequest, proxyInvocation.client()::createDBParameterGroup))
-                        .handleError((createDBParameterGroupRequest, exception, client, resourceModel, ctx) ->
-                                Commons.handleException(
-                                        ProgressEvent.progress(resourceModel, ctx),
-                                        exception,
-                                        DEFAULT_DB_PARAMETER_GROUP_ERROR_RULE_SET))
-                        .done((paramGroupRequest, paramGroupResponse, proxyInvocation, resourceModel, context) -> applyParameters(proxy, proxyInvocation, resourceModel, context, requestLogger)))
+
+        final ResourceModel desiredModel = request.getDesiredResourceState();
+
+        final Tagging.TagSet extraTags = Tagging.TagSet.builder()
+                .stackTags(Tagging.translateTagsToSdk(request.getDesiredResourceTags()))
+                .resourceTags(new HashSet<>(Translator.translateTagsToSdk(request.getDesiredResourceState().getTags())))
+                .build();
+
+        return ProgressEvent.progress(desiredModel, callbackContext)
+                .then(progress -> setDBParameterGroupNameIfEmpty(request, desiredModel, progress))
+                .then(progress -> createDBParameterGroup(proxy, proxyClient, progress, request.getSystemTags(), requestLogger))
+                .then(progress -> updateTags(proxyClient, progress, Tagging.TagSet.emptySet(), extraTags))
+                .then(progress -> applyParameters(proxy, proxyClient, progress.getResourceModel(), progress.getCallbackContext(), requestLogger))
                 .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, requestLogger));
     }
+
+    private ProgressEvent<ResourceModel, CallbackContext> createDBParameterGroup(final AmazonWebServicesClientProxy proxy,
+                                                                                 final ProxyClient<RdsClient> proxyClient,
+                                                                                 final ProgressEvent<ResourceModel, CallbackContext> progress,
+                                                                                 final Map<String, String> systemTags,
```

Comment:
We should abstract it from the specific tag set that we are fancy providing in this case.